arch/wasm32/wasm.js

```diff
@@ -24,14 +24,27 @@
 var heap_size_bytes = 16 * 1024 * 1024;
 var heap_size_pages = heap_size_bytes / (64 * 1024);
 var memory = new WebAssembly.Memory({initial: heap_size_pages, maximum: heap_size_pages})
-var heap = memory.buffer;
-var heap_uint8 = new Uint8Array(heap);
+var heap;
+var heap_uint8;
+var heap_uint32;
+
+function setHeap(h) {
+  heap = h
+  heap_uint8 = new Uint8Array(heap);
+  heap_uint32 = new Uint32Array(heap);
+  heap_size_bytes = heap.byteLength;
+}
+
+setHeap(memory.buffer)
```

Comment:
This will fail if code ever grows memory. Is that a concern for later? Should probably file an issue.

Reply:
You mean if memory ever grows? 

Yes, although that is true before this change too, no?

Reply:
Yes.